### PR TITLE
Provide better error message when using old version of Scanner for MS…

### LINKED ISSUE #489: Provide better error message when using old version of Scanner for MSBuild
### Repro steps

Run analysis with most recent SonarC# and a Scanner for MSBuild 2.1 or less.

### Expected behavior

An error stating the scanner version is not supported and the scanner needs to be upgraded. Perhaps with a link to how to downgrade C# plugin as a workaround.

### Actual behavior

Could not find a file on the SonarQube server. Url: http://sonarqubeserver/static/csharp/SonarQube.MSBuild.Runner.Implementation.zip

Failed to update the SonarQube Scanner for MSBuild binaries. Check the server url, verify that the C# plugin is correctly installed on the SonarQube server and that the SonarQube server has been restarted.

Pre-processing failed. Exit code: 1

ERROR: Execution of SonarQube Scanner for MSBuild failed (exit code 1)

### Known workarounds

- Downgrade C# plugin to 5.11 (build 1761).
- Upgrade the scanner to 2.2 or superior (may not always be an option).

### Related information

* SonarC# Version: 6.0+
* Scanner for MSBuild Version: 2.1 and older.

### PR DESCRIPTION
**Pull request by michalb-sonar**
…Build

Fix #489

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.DummyScanner/Program.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+
+namespace DummyScanner
+{
+    class Program
```
Comment (SonarTech) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.DummyScanner/Program.cs]: MAJOR Add a 'protected' constructor or the 'static' keyword to the class declaration. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1118)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: sonar-csharp-plugin/pom.xml [hunk 2]
```diff
@@ -215,6 +216,16 @@
                       </fileset>
                     </copy>
 
+                    <!-- These files are only used are only used by old scanners. Their only function is to produce a good error message and faild the build. -->
```
Comment (Evangelink) [inline sonar-csharp-plugin/pom.xml]: `are only used are only used`, you can remove one of them

Comment (Evangelink) [inline sonar-csharp-plugin/pom.xml]: `fail` and not `faild`

File: sonar-csharp-plugin/pom.xml [hunk 3]
```diff
@@ -215,6 +216,16 @@
...
+                    <copy file="${project.build.directory}/../../sonaranalyzer-dotnet/src/SonarAnalyzer.DummyScanner/bin/${analyzer.configuration}/DummyScanner.exe"
+                          tofile="${sonarAnalyzer.workDirectory}/MSBuild.Runner/SonarQube.MSBuild.PreProcessor.exe" />
+
```
Comment (Evangelink) [inline sonar-csharp-plugin/pom.xml]: I would remove the spaces to make it clear these tasks belong together. I don't whether there is any guideline on that I just know it is what I did everywhere else on this file.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.DummyScanner/Program.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+using System;
+
+namespace DummyScanner
+{
+    public static class Program
+    {
+        public static int Main(string[] args)
+        {
+            Console.Error.WriteLine("This version of scanner is deprecated. Upgrade to more recent version or downgrade SonarC# to pre-6.0 version.");
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.DummyScanner/Program.cs]: I would specify that they have to be at least on version 2.2.

**Comment by Evangelink** [comment]
@michalb-sonar Let me have a look for its. I will let you make one local test.

File: sonar-csharp-plugin/pom.xml (hunk 3 above)
Comment (michalb-sonar) [inline sonar-csharp-plugin/pom.xml]: That is the formatting used everywhere else in the file.

**Comment by Evangelink** [comment]
@michalb-sonar You can undo the changes related to adding a new project to `SonarAnalyzer.sln` after I wrote the IT I have realized we only need a zip (with a correct name) to exist. When you undo the changes you can merge.

**Pull request merged**